File: platform/jvm/gradle-test-app/src/main/java/io/bitdrift/gradletestapp/ui/theme/Theme.kt
// capture-sdk - bitdrift's client SDK
// Copyright Bitdrift, Inc. All rights reserved.
//
// Use of this source code is governed by a source available license that can be found in the
// LICENSE file or at:
// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt

package io.bitdrift.gradletestapp.ui.theme

import androidx.compose.material3.ColorScheme
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.darkColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color

private val AppColors: ColorScheme =
    darkColorScheme(
        primary = BitdriftColors.Primary,
        onPrimary = BitdriftColors.TextBright,
        background = BitdriftColors.Background,
        onBackground = BitdriftColors.TextPrimary,
        surface = BitdriftColors.BackgroundPaper,
        onSurface = BitdriftColors.TextPrimary,
        secondary = BitdriftColors.TextSecondary,
        onSecondary = BitdriftColors.TextBright,
        error = Color(0xFFCF6679),
    )

@Composable
fun BitdriftTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColors,
        typography = MaterialTheme.typography,
        shapes = MaterialTheme.shapes,
        content = content,
    )
}

File: examples/swift/hello_world/Theme.swift
// capture-sdk - bitdrift's client SDK
// Copyright Bitdrift, Inc. All rights reserved.
//
// Use of this source code is governed by a source available license that can be found in the
// LICENSE file or at:
// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt

import SwiftUI
import UIKit

struct Theme {
    static let primary = Color(hex: 0x00A76F)
    static let secondary = Color(hex: 0x8E33FF)
    static let background = Color(hex: 0x212B36)
    static let card = Color(hex: 0x919EAB, opacity: 0.2)
    static let surface = Color.white.opacity(0.05)
    static let border = Color.white.opacity(0.08)
    static let textPrimary = Color(hex: 0xFFFFFF)
    static let textSecondary = Color(hex: 0x919EAB)
    static let warning = Color(hex: 0xFFAB00)
    static let danger = Color(hex: 0xFF5630)

    static func applyNavigationAppearance() {
        let appearance = UINavigationBarAppearance()
        appearance.configureWithOpaqueBackground()
        appearance.backgroundColor = UIColor(Theme.background)
        appearance.shadowColor = .clear
        appearance.titleTextAttributes = [
            .foregroundColor: UIColor(Theme.textPrimary),
        ]
        appearance.largeTitleTextAttributes = [
            .foregroundColor: UIColor(Theme.textPrimary),
        ]

        UINavigationBar.appearance().standardAppearance = appearance
        UINavigationBar.appearance().scrollEdgeAppearance = appearance
        UINavigationBar.appearance().compactAppearance = appearance
        UINavigationBar.appearance().tintColor = UIColor(Theme.textPrimary)
    }
}

private extension Color {
    init(hex: UInt32, opacity: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255.0,
            green: Double((hex >> 8) & 0xFF) / 255.0,
            blue: Double(hex & 0xFF) / 255.0,
            opacity: opacity
        )
    }
}
